metricbeat/module/elasticsearch/index/index.go

```diff
@@ -59,34 +62,40 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 
 	isMaster, err := elasticsearch.IsMaster(m.HTTP, m.HostData().SanitizedURI+statsPath)
 	if err != nil {
-		r.Error(err)
+		err = errors.Wrap(err, "error determining if connected Elasticsearch node is master")
+		elastic.ReportAndLogError(err, r, m.Log)
 		return
 	}
 
 	// Not master, no event sent
 	if !isMaster {
-		logp.Debug("elasticsearch", "Trying to fetch index stats from a non master node.")
+		logp.Debug(elasticsearch.ModuleName, "Trying to fetch index stats from a non-master node.")
 		return
 	}
 
 	content, err := m.HTTP.FetchContent()
 	if err != nil {
-		r.Error(err)
+		elastic.ReportAndLogError(err, r, m.Log)
 		return
 	}
 
 	info, err := elasticsearch.GetInfo(m.HTTP, m.HostData().SanitizedURI)
 	if err != nil {
-		r.Error(err)
+		err = errors.Wrap(err, "failed to get info from Elasticsearch")
+		elastic.ReportAndLogError(err, r, m.Log)
 		return
 	}
 
 	if m.XPack {
-		eventsMappingXPack(r, m, *info, content)
+		err = eventsMappingXPack(r, m, *info, content)
+		if err != nil {
+			m.Log.Error(err)
```

Comment:
I definitely don't think Info, because then it would get logged all the time? I can see the argument both ways for Error vs. Debug. I'd rather keep it at Error because we only log it if something has gone wrong. Whereas Debug logging tends to happen regardless, just to provide additional detail.

Reply:
Why would info get logged all the time?